### PR TITLE
feat(password reset): generate new recovery key after reset

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - we want to automatically generate a new recovery key for the account when the previous one was used for a password reset so that the user does not need to do it manually

This commit:
 - generates a new recovery key and present it to the user if a recovery key was used during a password reset

### CODE REVIEW

**Review by vbudhram — APPROVED** [review]
@chenba This works as described. However, looks like the user will get a 3 emails after going through this flow, 1 for password reset complete, 1 for creating recovery key, 1 for automatic sign-in. Seems like we can consolidate or reduce the emails in some way, maybe a follow up?

**Comment by chenba** [comment]
> @chenba This works as described. However, looks like the user will get a 3 emails after going through this flow, 1 for password reset complete, 1 for creating recovery key, 1 for automatic sign-in. Seems like we can consolidate or reduce the emails in some way, maybe a follow up?

Good idea.  I'll file.  Thanks!

**Comment by chenba** [comment]
This is not working if the account has 2fa enabled.  Back to draft.

**Comment by chenba** [comment]
> users that have 2FA enabled shouldn't be auto logged in?

They are not. (And that's why this feature won't work for them.)

File: packages/functional-tests/tests/resetPassword/resetPasswordRecoveryKey.spec.ts [hunk 1]
```diff
@@ -62,13 +62,24 @@ test.describe('severity-1 #smoke', () => {
...
+      expect(filename.length).toBeGreaterThan(0);
+
+      await expect(resetPassword.recoveryKeyHintHeading).toBeVisible();
+      await resetPassword.recoveryKeyHintTextbox.fill('area 51');
+      await resetPassword.recoveryKeyFinishButton.click();
 
-      // TODO in FXA-7904 - Verify that a new recovery key is generated without needing to sign in again
-      // not currently implemented
-      await page.waitForURL(/settings\/account_recovery/);
+      await page.waitForURL(/settings/);
```
[the remark this answers is not in the document]
Comment (vpomerleau) [reply packages/functional-tests/tests/resetPassword/resetPasswordRecoveryKey.spec.ts:82]: Cleaning up the emails could be part of FXA-9544 that I'll be working on early next sprint - could add comments there about what to look out for/update?

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/en.ftl [hunk 2]
```diff
@@ -11,3 +11,8 @@ complete-reset-password-error-alert = Sorry, there was a problem setting your pa
 
 # Link to go back and use an account recovery key before resetting the password
 complete-reset-pw-recovery-key-link = Use account recovery key
+
+# A message informing the user that the password reset was successful and reminding them to create another recovery key
+# Displayed on the sign in page
+reset-password-complete-banner-heading = Your password has been reset.
+reset-password-complete-banner-message = Don’t forget to generate a new account recovery key from your { -product-mozilla-account } settings to prevent future sign-in issues.
```
Comment (chenba) [inline packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/en.ftl:18]: @bcolsson I had to add a couple more strings.  Mind taking a look?  Thanks!

**Comment by chenba** [comment]
I'm going to merge this since we want to do some testing on stage today.  I'll follow up if there are l10n concerns with the new strings.

**Pull request merged**